### PR TITLE
MSCO-22: add users to company directory

### PR DESCRIPTION
**Pull request by mavriciomart**
5.1 PR: https://github.com/2600hz/monster-ui-csv-onboarding/pull/54

### CODE REVIEW

File: app.js [hunk 1]
```diff
@@ -1230,6 +1256,100 @@ define(function(require) {
...
+					}
+				});
+
+				if (indexMain === -1) {
+					self.usersCreateMainDirectory(function(data) {
+						callback(data);
+					});
+				} else {
+					callback(listDirectories[indexMain]);
+				}
```
Comment (joristirado) [inline app.js]: This looks c/ped from some place else, but how about making it more declarative?

```js
var mainDirectory = _.find(listDirectories, { name: 'SmartPBX Directory' });

if (mainDirectory) {
    // mainDirectory exists
} else {
    // mainDirectory does not exists
}
```

**Review by joristirado — COMMENTED** [review]
Looks good!

Was there talks about giving the ability to opt-out from this, either as a csv column or a togglable input of some sort?

Only asking since Smart PBX gives that option during user creation. Also, that option is unset by default, just fyi.

File: app.js (hunk 1 above)
Comment (mavriciomart) [reply app.js]: Fixed. Thanks :)

**Comment by mavriciomart** [comment]
> Looks good!
> 
> Was there talks about giving the ability to opt-out from this, either as a csv column or a togglable input of some sort?
> 
> Only asking since Smart PBX gives that option during user creation. Also, that option is unset by default, just fyi.

@azefiel as per https://2600hz-commercial.atlassian.net/browse/MSCO-22?focusedCommentId=34062 I added a checkbox to provide the option to opt-in into this functionality.

File: app.js [hunk 2]
```diff
@@ -546,6 +550,7 @@ define(function(require) {
 
 			_.each(reviewData, function(userData) {
 				var newData = self.formatUserData(userData, data.customizations ? data.customizations : {});
+				newData.addUsersToMainDirectory = addUsersToMainDirectory;
```
Comment (joristirado) [inline app.js]: Mind moving this into the dedicated formatting method?

Obviously, the second argument has to be updated, so maybe pass the whole `data` variable?

File: i18n/en-US.json [hunk 3]
```diff
@@ -56,6 +56,7 @@
 				"others": "OTHERS"
 			},
 			"hasCustomizations": "Do you want to add JSON configs?",
+			"addToMainDirectory": "Do you want to add to Company Directory?",
```
Comment (joristirado) [inline i18n/en-US.json]:
```suggestion
			"addToMainDirectory": "Do you want to add users to Company Directory?",
```

Comment (mavriciomart) [reply i18n/en-US.json]: Fixed. Thanks!

File: app.js (hunk 2 above)
Comment (mavriciomart) [reply app.js]: Added a custom options argument to `formatUserData`

**Review by joristirado — APPROVED** [review]

**Pull request merged**